mars/services/task/supervisor/processor.py

```diff
@@ -72,6 +74,25 @@ async def inner(processor: "TaskProcessor", *args, **kwargs):
     return inner
 
 
+def _init_subtask_required_resource(subtask_graph: SubtaskGraph):
```

Comment:
No need to create a separate method.

Reply:
This method is designed to implement different initialization of subtasks' required resource. By default we just initialize  resources with special values, and like HBO, we can get the resources through external service.

Reply:
A bare modular method is not friendly for future maintainers. At least it should be a class member.

Reply:
I could add a 'ResourceAssigner' which assigns resources for subtasks by configuration, like default values or external service.

Reply:
Maybe `ResourceEvaluator` could be a better name.

Reply:
Good suggestion.